GitCommands/RevisionReader.cs

```diff
@@ -27,30 +28,37 @@ public sealed class RevisionReader
             /* Committer name  */ "%cN%n" +
             /* Committer email */ "%cE%n" +
             /* Reflog selector placeholder */ "{0}" +
-            /* Commit raw body */ "%B";
+            /* Commit raw body */ "%B" +
+            /* Notes placeholder */ "{1}";
 
         private const string _reflogSelectorFormat = "%gD%n";
+        private const string _notesPrefix = "Notes:";
+        private const string _notesMarker = $"\n{_notesPrefix}";
+        private const string _notesFormat = $"%n{_notesPrefix}%n%N";
 
         // Trace info for parse errors
         private int _noOfParseError = 0;
 
-        private readonly GitModule _module;
+        private readonly IGitModule _module;
         private readonly Encoding _logOutputEncoding;
         private readonly long _oldestBody;
         private const int _offsetDaysForOldestBody = 6 * 30; // about 6 months
 
         // reflog selector to identify stashes
         private bool _hasReflogSelector;
 
+        // Include Git Notes for the commit
+        private bool _hasNotes;
+
         // Buffer to decode subject
         private char[] _decodeBuffer = new char[4096];
 
-        public RevisionReader(GitModule module, bool allBodies = false)
+        public RevisionReader(IGitModule module, bool allBodies = false)
```

Comment:
Nice!
I have a draft where I change all signatures `GitModule` -> `IGitModule`, but it's in a way hinged on #11242 and the converstion of `HotkeySettingsManager` to an instance-based type. The worst part of that conversation is updating `IGitModule` with the missing definitions.